pipeline_dp/dp_engine.py

```diff
@@ -426,7 +427,7 @@ def _select_private_partitions_internal(self, col,
             collection of elements (partition_key, accumulator).
         """
         budget = self._budget_accountant.request_budget(
-            mechanism_type=pipeline_dp.MechanismType.GENERIC)
+            mechanism_type=mechanism_type)
```

Comment:
I realized that hereit should be more compicated :)

Laplace mechanism and Laplace PartitionSelection have different budget implications, I'll think and write down in more details

Reply:
I thought more, while it's not optimal from an accuracy point of view, it's simpler for now to use
`pipeline_dp.MechanismType.GENERIC`

i.e. let's consdier all strategies as (eps, delta)-dp mechanism.

**Explanation**: while Laplace/GaussianParittionSelectionStrategy use internally Laplace/Gaussian mechanism corresponding, it's done in a different way then classic usage Laplace/Gaussian mechasnism. For example any partition selection strategy should have `delta > 0` (because empty set should have probably 0, while set with 1 elements have probably > 0, so for such set only eps is not enough).